src/domain/gift-cards/new.tsx

```diff
@@ -0,0 +1,208 @@
+  component: React.ReactNode
+}
+
+const NewGiftCard: React.FC<NewGiftCardProps> = ({ onClose }) => {
+  const [uploading, setUploading] = useState(false)
+  const { register, setValue, unregister, handleSubmit } = useForm()
+  const { store } = useAdminStore()
+  const { refetch } = useAdminProducts()
+  const giftCard = useAdminCreateProduct()
+  const [denominations, setDenominations] = useState<Denomination[]>([])
+  const toaster = useToaster()
+
+  const handleValueUpdate = (name: string, value?: number) => {
+    if (!value) return
+    setValue(name, value)
+  }
+
+  const addDenomination = () => {
+    const name = `denominations.${denominations.length}`
+    const component = (
+      <CurrencyInput
+        currentCurrency={store?.default_currency_code}
+        readOnly
+        size="medium"
+      >
+        <CurrencyInput.AmountInput
+          label="Amount"
+          amount={undefined}
+          onChange={(v) => handleValueUpdate(name, v)}
+        />
+      </CurrencyInput>
+    )
+    register(name)
+    setDenominations([...denominations, { name, component }])
+  }
+
+  const deleteDenomination = (name: string) => {
+    unregister(name)
+    setDenominations(denominations.filter((d) => d.name !== name))
+  }
+
+  const handleFileUpload = async (files) => {
+    setUploading(true)
+    //TODO upload files
+    await new Promise((r) => setTimeout(r, 2000))
+    setUploading(false)
+  }
+
+  const onSubmit = (data: {
+    name: string
+    description?: string
+    denominations: number[]
+  }) => {
+    giftCard.mutate(
+      {
+        is_giftcard: true,
+        title: data.name,
+        description: data.description,
+        discountable: false,
+        options: [{ title: "Denominations" }],
+        variants: data.denominations.map((d, i) => ({
+          title: `${i + 1}`,
+          inventory_quantity: 0,
+          manage_inventory: false,
+          prices: [{ amount: d, currency_code: store?.default_currency_code }],
+          options: [{ value: `${d}` }],
+        })),
+        status: ProductStatus.PUBLISHED,
+      },
+      {
+        onSuccess: () => {
+          toaster("Successfully created gift card", "success")
+          refetch()
+          navigate("/a/gift-cards/manage")
+        },
+        onError: (err) => {
+          toaster(getErrorMessage(err), "error")
+        },
+      }
+    )
+  }
+
+  return (
+    <Modal handleClose={onClose}>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Modal.Body>
+          <Modal.Header handleClose={onClose}>
+            <div>
+              <h1 className="inter-xlarge-semibold">Create Gift Card</h1>
+            </div>
+          </Modal.Header>
+          <Modal.Content>
+            <div className="mb-base">
+              <h3 className="inter-base-semibold">Product information</h3>
+            </div>
+            <div className="flex flex-col gap-y-base">
+              <InputField
+                label={"Name"}
+                required
```

Comment:
This `required` is not enforced, the giftcard can be created without inputting anything